src/test/lrsql/admin/protocol_test.clj

```diff
@@ -155,7 +187,79 @@
         (is (-> (adp/-ensure-account-oidc lrs username bad-issuer)
                 :result
                 (= :lrsql.admin/oidc-issuer-mismatch-error)))))
-    (component/stop sys')))
+    (testing "actor-delete statements"
+      (let [stmts [stmt-0 stmt-1 stmt-2 stmt-3]
+            ifis (->> (conj stmts (stmt-3 "object"))
+                      (map #(ua/actor->ifi (% "actor"))))
+            get-actor-ss-count (fn [ifi]
+                                 (-> (lrsp/-get-statements lrs auth-ident {:actor-ifi ifi} [])
+                                     (get-in [:statement-result :statements])
+                                     count))
+            get-stmt-#s (fn []
+                          (reduce (fn [m actor-ifi]
+                                    (assoc m actor-ifi (get-actor-ss-count actor-ifi)))
+                                  {} ifis))]
+        (lrsp/-store-statements lrs auth-ident stmts [])
+
+        (is (every? #(> % 0) (vals (get-stmt-#s))))
+        (doseq [ifi ifis]
+          (adp/-delete-actor lrs {:actor-ifi ifi}))
+        (is (every? #(= % 0) (vals (get-stmt-#s))))))
+    (testing "actor delete and related statements"
+      (let [stmt->ifi #(ua/actor->ifi (% "actor"))
+            count-of-actor (fn [actor-ifi] (-> (lrsp/-get-statements lrs auth-ident {:actor-ifi actor-ifi} []) :statement-result :statements count))
+            child-ifi (stmt->ifi (stmt-3 "object"))
+            parent-ifi (stmt->ifi stmt-3)]
+        (testing "delete-actor correctly deletes statements that are parent to actor (sub)statements"
+          (lrsp/-store-statements lrs auth-ident [stmt-3] [])
+          (is (= 1 (count-of-actor parent-ifi)))
+          (adp/-delete-actor lrs {:actor-ifi child-ifi})
+          (is (zero? (count-of-actor parent-ifi))) ;
+          (adp/-delete-actor lrs {:actor-ifi parent-ifi}))
+        (testing "delete-actor correctly deletes substatements that are child to actor statements"
+          (lrsp/-store-statements lrs auth-ident [stmt-3] [])
+          (is (= 1 (count-of-actor child-ifi)))
+          (adp/-delete-actor lrs {:actor-ifi parent-ifi})
+          (is (zero? (count-of-actor child-ifi)))
+          (adp/-delete-actor lrs {:actor-ifi child-ifi}))
+        (testing "for StatementRefs, delete-actor deletes statement->actor relationships but leaves statements by another actor untouched"
+          (let [[ifi-0 ifi-2] (mapv stmt->ifi [stmt-0 stmt-2])]
+            (lrsp/-store-statements lrs auth-ident [stmt-0 stmt-2] [])
+            (is (= 2 (count-of-actor ifi-0)))
+            (is (= 2 (count-of-actor ifi-2)))
+            (adp/-delete-actor lrs {:actor-ifi ifi-0})
+            (is (= 1 (count-of-actor ifi-0)))
+            (is (= 1 (count-of-actor ifi-2)))
```

Comment:
Actually @cliffcaseyyet can you confirm that leaving deleted IFIs in statements is okay as long as they're referenced by a Statement Ref, right?

Reply:
@kelvinqian00 That's not exactly what is happening afaik. What we decided was if A (with Actor X) references B (with Actor Y) and we delete Actor Y, statement B gets deleted but Statement A remains in place with a dead ref (apparently legal in xAPI).